python/lsst/dax/ppdbx/gcp/db.py

```diff
@@ -0,0 +1,472 @@
+
+        Returns
+        -------
+        int
+            The number of rows inserted, which should be 1 if successful.
+        """
+        insert_values = {"apdb_replica_chunk": chunk_id, **values}
+        logging.info("Preparing to insert replica chunk %d with values: %s", chunk_id, insert_values)
+        stmt = insert(self.table).values(insert_values)
+        with self.engine.begin() as conn:
+            result = conn.execute(stmt)
+            affected_rows = result.rowcount
+
+        new_status = insert_values.get("status")
+        if affected_rows == 0:
+            logging.error(
+                "Expected to insert replica chunk %s with status '%s', but no rows were inserted.",
+                chunk_id,
+                new_status,
+            )
+            raise RuntimeError(
+                f"No rows inserted for apdb_replica_chunk={chunk_id} - insert silently failed."
+            )
+        else:
+            logging.info(
+                "Successfully inserted %d row(s) for replica chunk %s with status '%s'",
+                affected_rows,
+                chunk_id,
+                new_status,
+            )
+        return affected_rows
+
+    def get_promotable_chunks(self) -> list[tuple[int]]:
+        """
+        Return the first uninterrupted sequence of staged chunks such that all
+        prior chunks are promoted.
+
+        Returns
+        -------
+        list[tuple[int]]
+            A list of tuples containing the `apdb_replica_chunk` values of the
+            promotable chunks.
+        """
+        query = """
+        WITH start AS (
+        SELECT MIN(apdb_replica_chunk) AS s
+        FROM "PpdbReplicaChunk"
+        WHERE status <> 'promoted'
+        ),
+        stop AS (
+        SELECT MIN(p.apdb_replica_chunk) AS e
+        FROM "PpdbReplicaChunk" p
+        JOIN start ON TRUE
+        WHERE start.s IS NOT NULL
+            AND p.apdb_replica_chunk >= start.s
+            AND p.status <> 'staged'
+        )
+        SELECT p.apdb_replica_chunk
+        FROM "PpdbReplicaChunk" p
+        JOIN start ON TRUE
+        LEFT JOIN stop ON TRUE
+        WHERE start.s IS NOT NULL
+        AND p.status = 'staged'
+        AND p.apdb_replica_chunk >= start.s
+        AND (stop.e IS NULL OR p.apdb_replica_chunk < stop.e)
+        ORDER BY p.apdb_replica_chunk;
+        """
+        return self.execute(query)
+
+    def mark_chunks_promoted(self, promotable_chunks: list[tuple[int]]) -> int:
+        """Set status='promoted' for the given chunk IDs. Returns number
+        updated. Uses ``_db.execute(...)`` by wrapping UPDATE in a CTE that
+        returns a count.
+
+        Parameters
+        ----------
+        promotable_chunks : list[tuple[int]]
+            List of tuples containing the `apdb_replica_chunk` values of the
+            promotable chunks. Each tuple should contain a single integer
+            value.
+
+        Returns
+        -------
+        int
```

Comment:
```suggestion
        count : `int`
```
Please check all other docstrings and fix `Returns` to use names.